.github/workflows/ci.yml

```diff
@@ -581,6 +581,91 @@ jobs:
             gcr.io/iree-oss/base@sha256:605d86ccf4197e978a24867fabb7fc100334c926b067ee0518e46d0a4396e206 \
             ./build_tools/cmake/build_and_test_tsan.sh
 
+  small_runtime:
+    needs: setup
+    if: needs.setup.outputs.should-run == 'true'
+    runs-on: ubuntu-20.04-64core
```

Comment:
I was going to suggest grouping some of these jobs if the overhead of checking out the repo and its submodules exceeded the time to run the build, but it looks like those steps are fast enough on these machines (2s and 11s, then a 1 minute build)

Reply:
Yeah checkouts are way faster on GitHub runners. We should be able to optimize them on our runners too somehow. I'm actually surprised at how slow they are given that we're fetching with depth=1. I think I tried using our own submodule checkout with `--jobs=8`, but it didn't seem to help. Reference repos or local mirrors might help